website.js

```diff
@@ -319,6 +316,17 @@ io.on('connection', function (socket) {
             socket.emit('nextSong');
         }})
     })
+    socket.on('createPoll', function (data) {
+        connection.query('insert into poll set ?', data, function(err, result) {})
+    })
+    socket.on('getPoll', function (data) {
+        connection.query('select * from poll where id = ?', data, function(err, result) {
+            socket.emit('pollData', result[0].answers)
+        })
+    })
+    socket.on('addResult', function (data) {
+        connection.query('update poll set answers = ? where id = "' + data.id + '"', JSON.stringify(data.answers), function(err,result){})
```

Comment:
Your setup should probably look something like this:

#### Table 1: polls
| id | question |
|---|---------|
| 1  | "What should I eat for breakfast?" |
| 2 | "What is the next game I should play?" |

#### Table 2: answers
| id | poll_id | answer |
|---|--------|--------|
| 1 | 1 | "Cereal" |
| 2 | 1 | "Sandwich" |
| 3 | 2 | "League of Legends" |
| 4 | 1 | "Soup" |
| 5 | 2 | "CS:GO" |

#### Table 3: users
| id | username |
|---|-----------|
| 1 | justdrawcats |
| 2 | bobross |

#### Table 4: user_poll_answers (I don't know about the name though)
| id | poll_id | answer_id | user_id |
|---|--------|-----------|--------|
| 1 | 1 | 2 | 1 |

This way I could ask the database, _"give me all the answers justdrawcats has given for every question"_.

This also solves the issue where you are not able to check if the user already voted. You can now just do this (roughly):

```sql
SELECT id FROM user_poll_answers WHERE user_id = 1 AND poll_id = 1;
```

If the above query finds the right answer, then he already answered. If not, then you can actually save the answer 👍 . Feel free to hit me up if you have any questions though! It's a hard concept to understand at first, but it will get better over time!